playbooks/prepare_stack.yaml

```diff
@@ -27,16 +30,31 @@
     environment:
       OS_CLOUD: standalone
 
-  - name: Create public network
+  - name: Create external network
     shell: |
-      if ! openstack network show public; then
-          openstack network create --external --share --provider-physical-network datacentre --provider-network-type flat public
+      if ! openstack network show external; then
+          openstack network create --external --share --provider-physical-network external --provider-network-type flat external
       fi
-      if ! openstack subnet show public-subnet; then
-          openstack subnet create public-subnet --subnet-range "{{ public_cidr }}" \
-              --no-dhcp --gateway "{{ public_gateway }}" \
-              --allocation-pool "start={{ public_fip_pool_start }},end={{ public_fip_pool_end }}" \
-              --network public
+      if ! openstack subnet show external-subnet; then
+          openstack subnet create external-subnet --subnet-range "{{ external_cidr }}" \
+              --no-dhcp --gateway "{{ external_gateway }}" \
+              --allocation-pool "start={{ external_fip_pool_start }},end={{ external_fip_pool_end }}" \
+              --network external
+      fi
+    environment:
+      OS_CLOUD: standalone
+    when: external_fip_pool_start is defined and external_fip_pool_end is defined
+
+  - name: Create hostonly network
+    shell: |
+      if ! openstack network show hostonly; then
+          openstack network create --external --share --provider-physical-network hostonly --provider-network-type flat hostonly
+      fi
+      if ! openstack subnet show hostonly-subnet; then
+          openstack subnet create hostonly-subnet --subnet-range "{{ hostonly_cidr }}" \
+              --no-dhcp --gateway "{{ hostonly_gateway }}" \
```

Comment:
same here about dhcp.

Reply:
See comment above about enabling this manually.